### PR TITLE
PYIC-2302: Add new error response to state machine.

### PR DESCRIPTION
**Pull request by Danomanic**
### What changed
- Updated the State Machine to return a Error Response Object when CRI_ERROR returns.

### Issue tracking

- [PYIC-2302](https://govukverify.atlassian.net/browse/PYIC-2302)

[PYIC-2302]: https://govukverify.atlassian.net/browse/PYIC-2302?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/statemachine/responses/ErrorResponse.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public Map<String, Object> value(String id) {
+        // TODO: Come up with a better way of converting httpsStatusCode to an Integer.
+        return Map.of(
+                "type",
+                ERROR,
+                "pageId",
+                id,
+                "httpStatusCode",
+                httpStatusCode,
+                "statusCode",
```
Comment (dannyhayes-gds) [inline lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/statemachine/responses/ErrorResponse.java:31]: ~~Do we need this one? If we already return it as `httpStatusCode` as a string then not sure if we need another as an integer?~~

File: lambdas/process-journey-step/src/main/resources/statemachine/dev/ipv-core-main-journey.yaml [hunk 2]
```diff
@@ -15,8 +15,9 @@ CRI_STATE:
       name: error
       targetState: CRI_ERROR
       response:
-        type: page
+        type: error
```
Comment (dannyhayes-gds) [inline lambdas/process-journey-step/src/main/resources/statemachine/dev/ipv-core-main-journey.yaml]: I'd change this back for now until we stage it out with a feature flag

Comment (Danomanic) [reply lambdas/process-journey-step/src/main/resources/statemachine/dev/ipv-core-main-journey.yaml]: Done this, and `@Disabled` tests for the time being.

File: lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/statemachine/responses/ErrorResponse.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import uk.gov.di.ipv.core.library.service.ConfigurationService;
+
+import java.util.Map;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class ErrorResponse implements JourneyStepResponse {
+
+    public static final String ERROR = "error";
```
Comment (Danomanic) [inline lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/statemachine/responses/ErrorResponse.java]: should be private

Comment (Danomanic) [reply lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/statemachine/responses/ErrorResponse.java]: Fixed

**Review by dannyhayes-gds — APPROVED** [review]

**Pull request merged**